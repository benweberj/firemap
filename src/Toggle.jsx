const Toggle = (props) => {
  const { on, toggle } = props

  return (
    <div className={`toggle ${on && 'on'}`} onClick={toggle}>
      <div className={'toggle-ball'}></div>
    </div>
  )

}

export default Toggle