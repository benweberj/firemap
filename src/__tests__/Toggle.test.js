import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Toggle from '../Toggle';

describe('Toggle', () => {

    // Test to see if toggle renders darkmode
    it('Dark mode toggle renders correctly', () => {
      const { container } = render(<Toggle on={false} toggle={() => {}} />);
      // eslint-disable-next-line testing-library/no-node-access
      expect(container.firstChild).toMatchSnapshot();
    });
});