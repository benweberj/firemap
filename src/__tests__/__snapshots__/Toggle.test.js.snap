// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Dark mode toggle renders correctly 1`] = `
<div
  class="toggle false"
>
  <div
    class="toggle-ball"
  />
</div>
`;
